static/js/stream_data.js

```diff
@@ -459,20 +459,21 @@ exports.get_streams_for_admin = function () {
 };
 
 exports.initialize_from_page_params = function () {
-    function populate_subscriptions(subs, subscribed) {
+    function populate_subscriptions(subs, subscribed, previously_subscribed) {
         subs.forEach(function (sub) {
             var stream_name = sub.name;
             sub.subscribed = subscribed;
+            sub.previously_subscribed = previously_subscribed;
 
             exports.create_sub_from_server_data(stream_name, sub);
         });
     }
 
     exports.set_realm_default_streams(page_params.realm_default_streams);
 
-    populate_subscriptions(page_params.subscriptions, true);
-    populate_subscriptions(page_params.unsubscribed, false);
-    populate_subscriptions(page_params.never_subscribed, false);
+    populate_subscriptions(page_params.subscriptions, true, true);
+    populate_subscriptions(page_params.unsubscribed, false, true);
+    populate_subscriptions(page_params.never_subscribed, false, false);
```

Comment:
First, all current subscriber's value for `previously_subscribed` will set to `true`. 
Then subscribers which aren't currently subscribed(they can be never subscribed or previously subscribed but not currently subscribed users) value for `previously_subscribed` will set to `true`.
The value of `previously_subscribed` will set to `false` automatically overwritten on call of third line who are never subscribed users from unsubscribed users list(which we had previously set to `true`) to `false`.